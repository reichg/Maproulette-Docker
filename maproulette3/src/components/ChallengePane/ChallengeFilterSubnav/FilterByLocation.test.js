import React from 'react'
import { FilterByLocation } from './FilterByLocation'
import { cloneDeep as _cloneDeep } from 'lodash'

const propsFixture = {
  user: {
    id: 11,
    savedChallenges: [],
  },
  challenges: [
    {
      id: 309,
      name: "Challenge 309",
      blurb: "Challenge 309 blurb",
      description: "Challenge 309 description",
      parent: {
        displayName: "foo",
      }
    },
    {
      id: 311,
      name: "Challenge 311",
      blurb: "Challenge 311 blurb",
      description: "Challenge 311 description",
      parent: {
        displayName: "bar",
      }
    }
  ],
  searchFilters: {
    difficulty: 1
  },
  mapBounds: {
    challenges: {bounds: 1}
  },
}

let basicProps = null

beforeEach(() => {
  basicProps = _cloneDeep(propsFixture)
  basicProps.setSearchFilters = jest.fn()
  basicProps.removeSearchFilters = jest.fn()
  basicProps.locateMapToUser = jest.fn()
  basicProps.intl = {formatMessage: jest.fn()}
})

test("it renders with props as expected", () => {
  const wrapper = shallow(
    <FilterByLocation {...basicProps} />
  )

  expect(wrapper).toMatchSnapshot()
})

test("it calls setSearchFilters/locateMapToUser if an onChange occurs with 'nearMe'", () => {
  const wrapper = shallow(
    <FilterByLocation {...basicProps} />
  )

  wrapper.instance().updateFilter({value: 'nearMe'})
  expect(basicProps.setSearchFilters).toBeCalledWith({"location": "withinMapBounds"})
  expect(basicProps.locateMapToUser).toBeCalledWith({"id": 11, "savedChallenges": []})
})


test("it calls setSearchFilters if an onChange occurs with withinMapBounds", () => {
  const wrapper = shallow(
    <FilterByLocation {...basicProps} />
  )

  wrapper.instance().updateFilter({value: 'withinMapBounds'})
  expect(basicProps.setSearchFilters).toBeCalledWith({"location": "withinMapBounds"})
})

test("it calls removeSearchFilters if an onChange occurs with a null value", () => {
  const wrapper = shallow(
    <FilterByLocation {...basicProps} />
  )

  wrapper.instance().updateFilter({value: null})
  expect(basicProps.removeSearchFilters).toBeCalledWith(['location'])
})
