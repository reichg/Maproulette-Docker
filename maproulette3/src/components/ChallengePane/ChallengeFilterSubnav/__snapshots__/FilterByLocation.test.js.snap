// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders with props as expected 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FilterByLocation
    challenges={
      Array [
        Object {
          "blurb": "Challenge 309 blurb",
          "description": "Challenge 309 description",
          "id": 309,
          "name": "Challenge 309",
          "parent": Object {
            "displayName": "foo",
          },
        },
        Object {
          "blurb": "Challenge 311 blurb",
          "description": "Challenge 311 description",
          "id": 311,
          "name": "Challenge 311",
          "parent": Object {
            "displayName": "bar",
          },
        },
      ]
    }
    intl={
      Object {
        "formatMessage": [MockFunction] {
          "calls": Array [
            Array [
              Object {
                "defaultMessage": "Near Me",
                "id": "Challenge.location.nearMe",
              },
            ],
            Array [
              Object {
                "defaultMessage": "Within Map Bounds",
                "id": "Challenge.location.withinMapBounds",
              },
            ],
            Array [
              Object {
                "defaultMessage": "Intersecting Map Bounds",
                "id": "Challenge.location.intersectingMapBounds",
              },
            ],
            Array [
              Object {
                "defaultMessage": "Anywhere",
                "id": "Challenge.location.any",
              },
            ],
          ],
          "results": Array [
            Object {
              "isThrow": false,
              "value": undefined,
            },
            Object {
              "isThrow": false,
              "value": undefined,
            },
            Object {
              "isThrow": false,
              "value": undefined,
            },
            Object {
              "isThrow": false,
              "value": undefined,
            },
          ],
        },
      }
    }
    locateMapToUser={[MockFunction]}
    mapBounds={
      Object {
        "challenges": Object {
          "bounds": 1,
        },
      }
    }
    removeSearchFilters={[MockFunction]}
    searchFilters={
      Object {
        "difficulty": 1,
      }
    }
    setSearchFilters={[MockFunction]}
    user={
      Object {
        "id": 11,
        "savedChallenges": Array [],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "button": <ButtonFilter
        type={
          <FormattedMessage
            defaultMessage="Location"
            id="ChallengeFilterSubnav.filter.location.label"
            values={Object {}}
          />
        }
      />,
      "children": <ol
        className="mr-list-dropdown mr-list-dropdown--ruled"
      >
        <li>
          <a
            onClick={[Function]}
          />
        </li>
        <li>
          <a
            onClick={[Function]}
          />
        </li>
        <li>
          <a
            onClick={[Function]}
          />
        </li>
        <li>
          <a
            onClick={[Function]}
          />
        </li>
      </ol>,
      "className": "mr-dropdown--flush xl:mr-border-l xl:mr-border-white-10 mr-p-6 mr-pl-0 xl:mr-pl-6",
      "close": [Function],
      "isVisible": false,
      "toggleVisible": [Function],
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <li>
            <a
              onClick={[Function]}
            />
          </li>,
          <li>
            <a
              onClick={[Function]}
            />
          </li>,
          <li>
            <a
              onClick={[Function]}
            />
          </li>,
          <li>
            <a
              onClick={[Function]}
            />
          </li>,
        ],
        "className": "mr-list-dropdown mr-list-dropdown--ruled",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "any",
          "nodeType": "host",
          "props": Object {
            "children": <a
              onClick={[Function]}
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
              "onClick": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": "a",
          },
          "type": "li",
        },
        Object {
          "instance": null,
          "key": "withinMapBounds",
          "nodeType": "host",
          "props": Object {
            "children": <a
              onClick={[Function]}
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
              "onClick": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": "a",
          },
          "type": "li",
        },
        Object {
          "instance": null,
          "key": "intersectingMapBounds",
          "nodeType": "host",
          "props": Object {
            "children": <a
              onClick={[Function]}
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
              "onClick": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": "a",
          },
          "type": "li",
        },
        Object {
          "instance": null,
          "key": "nearMe",
          "nodeType": "host",
          "props": Object {
            "children": <a
              onClick={[Function]}
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
              "onClick": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": "a",
          },
          "type": "li",
        },
      ],
      "type": "ol",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "button": <ButtonFilter
          type={
            <FormattedMessage
              defaultMessage="Location"
              id="ChallengeFilterSubnav.filter.location.label"
              values={Object {}}
            />
          }
        />,
        "children": <ol
          className="mr-list-dropdown mr-list-dropdown--ruled"
        >
          <li>
            <a
              onClick={[Function]}
            />
          </li>
          <li>
            <a
              onClick={[Function]}
            />
          </li>
          <li>
            <a
              onClick={[Function]}
            />
          </li>
          <li>
            <a
              onClick={[Function]}
            />
          </li>
        </ol>,
        "className": "mr-dropdown--flush xl:mr-border-l xl:mr-border-white-10 mr-p-6 mr-pl-0 xl:mr-pl-6",
        "close": [Function],
        "isVisible": false,
        "toggleVisible": [Function],
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li>
              <a
                onClick={[Function]}
              />
            </li>,
            <li>
              <a
                onClick={[Function]}
              />
            </li>,
            <li>
              <a
                onClick={[Function]}
              />
            </li>,
            <li>
              <a
                onClick={[Function]}
              />
            </li>,
          ],
          "className": "mr-list-dropdown mr-list-dropdown--ruled",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "any",
            "nodeType": "host",
            "props": Object {
              "children": <a
                onClick={[Function]}
              />,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
                "onClick": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": "a",
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": "withinMapBounds",
            "nodeType": "host",
            "props": Object {
              "children": <a
                onClick={[Function]}
              />,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
                "onClick": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": "a",
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": "intersectingMapBounds",
            "nodeType": "host",
            "props": Object {
              "children": <a
                onClick={[Function]}
              />,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
                "onClick": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": "a",
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": "nearMe",
            "nodeType": "host",
            "props": Object {
              "children": <a
                onClick={[Function]}
              />,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
                "onClick": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": "a",
            },
            "type": "li",
          },
        ],
        "type": "ol",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
